instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -89,6 +89,11 @@ def untraced_request?(env)
             @untraced_endpoints.include?(env['PATH_INFO'])
           end
 
+          def filtered_request?(env)
+            filter = config[:filtered_requests]
+            filter ? filter.call(env) : false
+          end
```

Comment:
I'd combine these two:
```suggestion
            return true if @untraced_endpoints.include?(env['PATH_INFO'])
            return true if config[:filtered_requests]&.call(env)
            
            false
          end
```